Doc: Updated Azure AD OAuth2 authentication topic. (#39609)
grafana/grafana: docs/sources/auth/azuread.md

- [Create the Azure AD application](#create-the-azure-ad-application)
- [Enable Azure AD OAuth in Grafana](/#enable-azure-ad-oauth-in-grafana)

## Create the Azure AD application

To enable the Azure AD OAuth2, register your application with Azure AD.

1. Log in to [Azure Portal](https://portal.azure.com), then click **Azure Active Directory** in the side menu. 

1. If you have access to more than one tenant, select your account in the upper right. Set your session to the Azure AD tenant you wish to use.

1. Under **Manage** in the side menu, click **App Registrations** > **New Registration**. Enter a descriptive name.

1. Under **Redirect URI**, select the app type **Web**.

1. Add the following redirect URLs `https://<grafana domain>/login/azuread` and `https://<grafana domain>` then click **Register**. The app's **Overview** page opens. 

1. Note the **Application ID**. This is the OAuth client ID.

1. Click **Endpoints** from the top menu. 

> 	- Note down the **OAuth 2.0 authorization endpoint (v2)** URL. This is the auth URL.
> 	- Note the **OAuth 2.0 token endpoint (v2)**. This is the token URL.

Reviewer: ```suggestion
	- Note the **OAuth 2.0 authorization endpoint (v2)** URL. This is the authorization URL.
	- Note the **OAuth 2.0 token endpoint (v2)**. This is the token URL.
```

Author: OMG! TY :)

> 1. Click **Certificates & secrets** and then add a new entry under **Client secrets** with the following configuration.

Reviewer: ```suggestion
1. Click **Certificates & secrets**, then add a new entry under **Client secrets** with the following configuration.
```

   - Description: Grafana OAuth
   - Expires: Never

1. Click **Add** then copy the key value. This is the OAuth client secret.

> 1. Click **Manifest**, then define the required Application Role values for Grafana: Viewer, Editor, Admin. Otherwise, all users will have the Viewer role. 

Reviewer: ```suggestion
1. Click **Manifest**, then define the required Application Role values for Grafana: Viewer, Editor, or Admin. If not defined, all users will have the Viewer role. 
```
